GameBoard.cpp

```diff
@@ -51,46 +59,104 @@ uint32_t GameBoard::xyToIndex(int x, int y) {
 
 }
 
-bool GameBoard::nextPointStatus(int x, int y) {
-	// TODO
-	return false;
-}
-
 uint8_t GameBoard::countNeighbors(uint32_t x, uint32_t y) {
 	// Would it be faster or slower to check if the value is 4 to avoid checking all pixels
 	// The if statements could lead to a slower processing time
 
 	// counts neighbors row by row
 	uint8_t count = 0;
 	uint32_t leftPoint = xyToIndex(x - 1, y - 1);
-	count += m_points.get(leftPoint);
-	count += m_points.get(leftPoint + 1);
-	count += m_points.get(leftPoint + 2);
+	count += m_cells.get(leftPoint);
+	count += m_cells.get(leftPoint + 1);
+	count += m_cells.get(leftPoint + 2);
 	leftPoint += c_sideLength;
-	count += m_points.get(leftPoint);
-	count += m_points.get(leftPoint + 2);
+	count += m_cells.get(leftPoint);
+	count += m_cells.get(leftPoint + 2);
 	leftPoint += c_sideLength;
-	count += m_points.get(leftPoint);
-	count += m_points.get(leftPoint + 1);
-	return count + m_points.get(leftPoint + 2);
+	count += m_cells.get(leftPoint);
+	count += m_cells.get(leftPoint + 1);
+	return count + m_cells.get(leftPoint + 2);
 }
 
 /*
 Chunk method definitions
 */
 
-GameBoard::Chunk::Chunk(GameBoard& gb, uint32_t offsetX, uint32_t offsetY) :
+GameBoard::Chunk::Chunk(GameBoard& gb, const Varient& varient, uint32_t offsetX, uint32_t offsetY) :
 	m_gb(gb),
 	c_offsetX(offsetX),
-	c_offsetY(offsetY) {}
+	c_offsetY(offsetY),
+	m_varient(varient),
+	m_border(BitArray(gb.c_chunkWidth * 2)) {
+	std::cout << "X: " << offsetX << " Y: " << offsetY << " test: " << m_varient.ySide << ' ' << m_varient.xSide << '\n';
+}
 
-// TODO: is this needed, or is Chunk::nextPointStatus sufficient
-uint8_t GameBoard::Chunk::countNeighbors(uint32_t x, uint32_t y) {
+bool GameBoard::Chunk::calcNextCellStatus(int x, int y) {
 	x += c_offsetX; y += c_offsetY;
-	return m_gb.countNeighbors(x, y);
+	bool alive = m_gb.getPoint(x, y);
+	uint8_t neighbors = m_gb.countNeighbors(x, y);
+
+	return neighbors == 3 || (alive && neighbors == 2);
 }
 
-bool GameBoard::Chunk::nextPointStatus(int x, int y) {
-	x += c_offsetX; y += c_offsetY;
-	return m_gb.nextPointStatus(x, y);
+void GameBoard::Chunk::processChunk() {
+	BitArray rows[2] = { BitArray(m_gb.c_chunkWidth), BitArray(m_gb.c_chunkWidth) };
+	bool currRow = true;

```

Comment:
Using bool for this is probably bad practice, but you were blowing my sleepy mind for a second with the toggling. My suggestion in the other comment uses XOR to flip an integer back and forth from 1 to 0 instead.

Reply:
That would make more sense, my brain was like I need a 0 and a 1. So I just went with bool.